hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -1127,6 +1128,90 @@ onOpenChainCloseTx openState newChainState closedSnapshotNumber contestationDead
 
   OpenState{parameters = headParameters, headId, coordinatedHeadState} = openState
 
+-- | Client request to side load confirmed snapshot.
+--
+-- Note this is not covered by the spec as it is not reachable from an organic use of the protocol.
+--
+-- It must not have any effects outside of a neutral modification of the state to:
+-- * something it was before (in the case of the initial snapshot).
+-- * something it would be using side communication (in the case of a confirmed snapshot).
+--
+-- Besides the above, it is expected to work very much like the confirmed snapshot.
+--
+-- __Transition__: 'OpenState' → 'OpenState'
+onOpenClientSideLoadSnapshot :: IsTx tx => OpenState tx -> ConfirmedSnapshot tx -> Outcome tx
+onOpenClientSideLoadSnapshot openState requestedConfirmedSnapshot =
+  case requestedConfirmedSnapshot of
+    InitialSnapshot{} ->
+      requireVerifiedSameSnapshot $
+        newState ClearLocalState{headId, snapshotNumber = lastSeenSn}
+    ConfirmedSnapshot{snapshot, signatures} ->
+      requireVerifiedSnapshotNumber $
+        requireVerifiedL1Snapshot $
+          requireVerifiedMultisignature snapshot signatures $
+            changes
+              [ SnapshotConfirmed{headId, snapshot, signatures}
+              , ClearLocalState{headId, snapshotNumber = lastSeenSn}
+              ]
+ where
+  OpenState
+    { headId
+    , parameters = HeadParameters{parties}
+    , coordinatedHeadState
+    } = openState
+
+  CoordinatedHeadState
+    { confirmedSnapshot = currentConfirmedSnapshot
+    } = coordinatedHeadState
+
+  vkeys = vkey <$> parties
+
+  currentSnapshot@Snapshot
+    { version = lastSeenSv
+    , number = lastSeenSn
+    , utxoToCommit = lastSeenSc
+    , utxoToDecommit = lastSeenSd
+    } = getSnapshot currentConfirmedSnapshot
+
+  requestedSnapshot@Snapshot
+    { version = requestedSv
+    , number = requestedSnapshotNumber
+    , utxoToCommit = requestedSc
+    , utxoToDecommit = requestedSd
+    } = getSnapshot requestedConfirmedSnapshot
+
+  requireVerifiedSameSnapshot cont =
+    if requestedSnapshot == currentSnapshot
+      then cont
+      else Error $ AssertionFailed "InitalSnapshot side loaded does not match last known."
+
+  requireVerifiedSnapshotNumber cont =
+    if requestedSnapshotNumber >= lastSeenSn
+      then cont
+      else
+        Error $
+          RequireFailed $
+            ReqSnNumberInvalid{requestedSn = requestedSnapshotNumber, lastSeenSn}
+
+  requireVerifiedL1Snapshot
+    cont
+      | requestedSv /= lastSeenSv = Error $ RequireFailed $ ReqSvNumberInvalid{requestedSv, lastSeenSv}
+      | requestedSc /= lastSeenSc = Error $ AssertionFailed "ConfirmedSnapshot utxoToCommit side loaded does not match last known."
+      | requestedSd /= lastSeenSd = Error $ AssertionFailed "ConfirmedSnapshot utxoToDecommit side loaded does not match last known."
+      | otherwise = cont
```

Comment:
Could: improve indentation here

I needed to read this function 2-3 times as it seemed at first that `cont` is an expression, but in fact it's an argument to the function.

```suggestion
  requireVerifiedL1Snapshot cont
      | requestedSv /= lastSeenSv = Error $ RequireFailed $ ReqSvNumberInvalid{requestedSv, lastSeenSv}
      | requestedSc /= lastSeenSc = Error $ AssertionFailed "ConfirmedSnapshot utxoToCommit side loaded does not match last known."
      | requestedSd /= lastSeenSd = Error $ AssertionFailed "ConfirmedSnapshot utxoToDecommit side loaded does not match last known."
      | otherwise = cont
```